src/AutoRest.CSharp.V3/ClientModels/ClientMethodPaging.cs

```diff
@@ -0,0 +1,27 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+namespace AutoRest.CSharp.V3.ClientModels
+{
+    internal class ClientMethodPaging
+    {
+        public ClientMethodPaging(ClientMethod method, ClientMethod nextPageMethod, string name, string? nextLinkName, string itemName, ClientTypeReference itemType, string? operationName)
+        {
+            Method = method;
+            NextPageMethod = nextPageMethod;
+            Name = name;
+            NextLinkName = nextLinkName;
+            ItemName = itemName;
+            ItemType = itemType;
+            OperationName = operationName;
+        }
+
+        public string Name { get; }
+        public ClientMethod Method { get; }
+        public ClientMethod NextPageMethod { get; }
+        public string? NextLinkName { get; }
+        public string ItemName { get; }
+        public ClientTypeReference ItemType { get; }
+        public string? OperationName { get; }
```

Comment:
You shouldn't need this considering we have NextPageMethod  reference

Reply:
NextPageMethod doesn't have to do with these values. They are created from the extension information from the original method. That makes sense to do because I have to determine if the extension exists ahead of time to know the operation is intended to be a pageable operation. Since I already have the pageable dictionary, we calculate those values while creating this type. It is a similar pattern to the other types of this sort.
![image](https://user-images.githubusercontent.com/17788297/72377184-20342680-36c4-11ea-82b0-c78c77258fbd.png)

Reply:
I was only talking about OperationName. I don't think it's used right now or would ever be used.

Reply:
Ah. Conceptually, it will be replaced with something like Operation, which would link to an actual operation. It doesn't hurt to be there and shows a clear path for implementing https://github.com/Azure/autorest.csharp/issues/397. Likely, a bunch of this code will change anyway as x-ms-pageable will become actual data in the code model via https://github.com/Azure/autorest.modelerfour/issues/85. So, I'm not hung up on the accuracy of the data we're using/storing right now. Putting extra information in doesn't hurt if it is intended to help us later. I'm sure we'll both want to do more cleanup as the project progresses.

Reply:
But we already have concept for this it's the `NextPageMethod` reference.